HW_Type_Inference.ts

```diff
@@ -0,0 +1,62 @@
+enum GridFilterTypeEnum {
+    MATCH,
+    RANGE,
+    VALUE_SET
+}
+
+type GridFilterMatch<T> = {
+    type: GridFilterTypeEnum.MATCH;
+    filter: Extract<T, string | number>;
+};
+
+type GridFilterRange<T> = {
+    type: GridFilterTypeEnum.RANGE;
+    filter: Extract<T, string | number>;
+    filterTo: Extract<T, string | number>;
+};
+
+type GridFilterValueSet<T> = {
+    type: GridFilterTypeEnum.VALUE_SET;
+    values: T[];
+};
+
+type GridFilterValue<T> = GridFilterMatch<T> | GridFilterRange<T> | GridFilterValueSet<T>;
+
+interface Movie {
+    title: string;
+    releaseYear: number;
+    rating: number;
+    awards: string[];
+}
+
+class MovieList {
```

Comment:
The first task requirement tells us that there are two entities. I cannot find the second one. Please double-check.